HHMI hub

GeorgianaElena opened the pull request:
For https://github.com/2i2c-org/infrastructure/issues/3080

This currently only has a staging hub, deployed at https://staging.hhmi.2i2c.cloud/.
Need to add a prod hub for it

consideRatio reviewed (commented):
This LGTM and consider this approved if you provide oauth 15.1 functional config now that the oauth16 upgrade was reverted. Work towards oauth16 can be found in https://github.com/2i2c-org/infrastructure/pull/3144

yuvipanda commented on config/clusters/hhmi/common.values.yaml:
```diff
@@ -0,0 +1,66 @@
+      cpu:
+        guarantee: 0.5
+        # We're on n1-highmem-16 machines
+        limit: 14
```
This probably needs a G after

GeorgianaElena replied:
@yuvipanda, I think it's ok because this is the cpu limit. I just double-checked and the memory guarantee and limit have the G.

GeorgianaElena commented:
Thanks @consideRatio and @yuvipanda for the review. I've now also added (and deployed) a prod hub too. Also updated the auth config for oauthenticator < 16

> This LGTM and consider this approved if you provide oauth 15.1 functional config

Per Erik's comment above, I will merge this now.

damianavila commented on terraform/gcp/projects/daskhub-template.tfvars:
```diff
@@ -8,7 +8,7 @@
 project_id = "{{ project_id }}"
 
-zone   = "{{ cluster_region }}"
+zone   = "{{ cluster_region }}-b"
```
Quick question in a high-level post review (so I might be totally wrong :wink:): do we always want to deploy to the `-b` zone? Should not get the `zone` from `foo_cluster.tfvars` file instead of "building it" from the region?
cc @GeorgianaElena

GeorgianaElena replied:
> Should not get the zone from foo_cluster.tfvars file instead of "building it" from the region?

@damianavila, so this template is actually used by the deployer to generate a `foo_cluster.tfvars` file. This initial file generated by the deployer is more of a starting point with some "default" basics that are meant to be edited to fulfill all of the needs for the cluster.

> do we always want to deploy to the -b zone?

Not necessarily, but it is my understanding that this is the go-to zone for us-west2 (which I believe is the default region if none is requested) if we want to be able to have GPU machines (which for a daskhub is probable) available and a vaster CPU types machines, including AMDs.

damianavila replied:
>@damianavila, so this template is 

I knew I was missing something (well-placed "so I might be totally wrong 😉" message).
Thanks for the clarification!!

> Not necessarily,

Yep, it makes sense now that I actually realize what I am seeing 😉
Again, thanks for your patience and feedback.

colliand commented:
I believe this issue can be closed since the hub has been deployed.
